src/audio/dai.c

```diff
@@ -688,6 +688,81 @@ static int dai_config(struct comp_dev *dev, struct sof_ipc_dai_config *config)
 	return 0;
 }
 
+static int dai_ctrl_set_cmd(struct comp_dev *dev,
+			    struct sof_ipc_ctrl_data *cdata)
+{
+	struct dai_data *dd = comp_get_drvdata(dev);
+	int ret = 0;
+	int val;
+
+	/* validate */
+	if (cdata->num_elems != 1) {
+		trace_dai_error("xs0");
+		return -EINVAL;
+	}
+
+	switch (cdata->cmd) {
+	case SOF_CTRL_CMD_SWITCH:
+		val = cdata->compv[0].uvalue;
+		trace_dai("dcs");
+		trace_value(cdata->comp_id);
```

Comment:
This will flow easier when using ioctl as the cmd can be passed directly in most cases.